pkgs/development/libraries/science/math/tensorrt/extension.nix

```diff
@@ -18,14 +19,20 @@ final: prev: let
     # Return the first file that is supported. In practice there should only ever be one anyway.
     supportedFile = files: findFirst isSupported null files;
     # Supported versions with versions as keys and file as value
-    supportedVersions = filterAttrs (version: file: file !=null ) (mapAttrs (version: files: supportedFile files) tensorRTVersions);
+    supportedVersions = filterAttrs (version: file: file != null) (mapAttrs (version: files: supportedFile files) tensorRTVersions);
     # Compute versioned attribute name to be used in this package set
     computeName = version: "tensorrt_${toUnderscore version}";
     # Add all supported builds as attributes
-    allBuilds = mapAttrs' (version: file: nameValuePair (computeName version) (buildTensorRTPackage (removeAttrs file ["fileVersionCuda"]))) supportedVersions;
+    allBuilds = mapAttrs' (version: file: nameValuePair (computeName version) (buildTensorRTPackage (removeAttrs file [ "fileVersionCuda" ]))) supportedVersions;
     # Set the default attributes, e.g. tensorrt = tensorrt_8_4;
-    defaultBuild = { "tensorrt" = allBuilds.${computeName tensorRTDefaultVersion}; };
-  in allBuilds // defaultBuild;
+    defaultBuildName = computeName tensorRTDefaultVersion;
+    defaultBuild =
+      if
+        builtins.hasAttr defaultBuildName allBuilds then
```

Comment:
could use the `?` operator

Reply:
I thought so too, but then I realized I'd have to use it with an interpolation: `? ${defaultBuildName}`. Wasn't sure if that's better than `hasAttr`